bots/hgbridge/src/test/java/org/openjdk/skara/bots/hgbridge/JBridgeBotFactoryTest.java

```diff
@@ -0,0 +1,72 @@
+import org.openjdk.skara.test.TestHostedRepository;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JBridgeBotFactoryTest {
+    @Test
+    public void testCreate() {
+        try (var tempFolder = new TemporaryDirectory()) {
+            var marks = new TestHostedRepository(null, "marks");
+            var converter = new TestHostedRepository(null, "converter");
+            var des1 = new TestHostedRepository(null, "des1");
+            var des2 = new TestHostedRepository(null, "des2");
+
+            String jsonString = """
+                    {
+                      "marks": {
+                        "repository": "marks",
+                        "ref": "master",
+                        "name": "test",
+                        "email": "test@test.org"
+                      },
+                      "converters": [
+                        {
+                          "repository": "converter",
+                          "ref": "master",
+                          "authors": "test_authors.json",
+                          "contributors": "test_contributors.json",
+                          "sponsors": "test_sponsors.json",
+                          "corrections": "test_corrections.json",
+                          "replacements": "test_replacements.json",
+                          "repositories": [
+                            {
+                              "source": "https://test.org/source1",
+                              "destinations": "https://test.org/des1",
+                              "replacements": "test_replacements_for_spec_repo.json",
+                            },
+                            {
+                              "source": "https://test.org/source2",
+                              "destinations": "https://test.org/des2"
+                            }
+                          ]
+                        }
+                      ]
+                    }
+                    """;
+            var jsonConfig = JWCC.parse(jsonString).asObject();
+
+            var testBotFactory = TestBotFactory.newBuilder()
+                    .addHostedRepository("marks", marks)
+                    .addHostedRepository("converter", converter)
+                    .addHostedRepository("https://test.org/des1", des1)
+                    .addHostedRepository("https://test.org/des2", des2)
+                    .storagePath(tempFolder.path().resolve("storage"))
+                    .build();
+
+            var bots = testBotFactory.createBots(JBridgeBotFactory.NAME, jsonConfig);
+            // A JBridgeBot for every configured repo
+            assertEquals(2, bots.size());
+
+            JBridgeBot jBridgeBot1 = (JBridgeBot) bots.get(0);
+            assertEquals("JBridgeBot@https://test.org/source1", jBridgeBot1.toString());
+            JBridgeBot jBridgeBot2 = (JBridgeBot) bots.get(1);
+            assertEquals("JBridgeBot@https://test.org/source2", jBridgeBot2.toString());
```

Comment:
I think this bot is a good example of where we should probably add some accessors to be able to verify the configuration in more detail.